client/src/components/checkout-form.js

```diff
@@ -0,0 +1,158 @@
+  }
+
+  constructor() {
+    super();
+    this.onSubmit = () => {};
+    this.submitForm = this.submitForm.bind(this);
+
+    this.state = {
+      openFormErrorDialog: false,
+      openSuccessDialog: false,
+      openErrorDialog: false,
+    };
+  }
+
+  toggleFormErrorDialog() {
+    this.state.openFormErrorDialog = !this.state.openFormErrorDialog;
+    this.requestUpdate();
+  }
+
+  toggleSuccessDialog() {
+    this.state.openSuccessDialog = !this.state.openSuccessDialog;
+    this.requestUpdate();
+  }
+
+  toggleErrorDialog() {
+    this.state.openErrorDialog = !this.state.openErrorDialog;
+    this.requestUpdate();
+  }
+
+  async submitForm(event) {
+    event?.preventDefault();
+    const form = new FormData(this.shadowRoot.querySelector('form') || {});
+    let response;
+
+    if (this.isValidEmail(form.get('email'))) {
+      try {
+        response = await this.onSubmit(form);
+
+        // TODO: getting called to earlier, moving up to parent checkout page to handle success/error handling
+        response?.status?.ok
+          ? this.toggleSuccessDialog()
+          : this.toggleErrorDialog();
+      } catch (error) {
+        console.error(error);
+      }
+    } else {
+      this.toggleFormErrorDialog();
+    }
+  }
+
+  isValidEmail(text) {
+    return /[^ @]*@[^ @]*/.test(text);
+  }
+
+  render() {
+    const { openFormErrorDialog, openSuccessDialog, openErrorDialog } =
+      this.state;
+
+    return html` <div>
+      <form @submit=${this.submitForm}>
+        <div class="formControls">
+          <mwc-textfield
+            outlined
+            required
+            autoValidate
+            name="email"
+            helper="foo@bar.com"
+            pattern="[^ @]*@[^ @]*"
+            placeholder="Enter your email"
+            validationMessage="Field is required"
+          ></mwc-textfield>
+          <mwc-select
+            name="type"
+            label="Payment Type"
+            required
+            validationMessage="Field is required"
+          >
+            <mwc-list-item></mwc-list-item>
+            <mwc-list-item selected value="credit">Credit</mwc-list-item>
+            <mwc-list-item value="collect">Collect</mwc-list-item>
+          </mwc-select>
+          <button type="submit" class="checkoutButton">Purchase</button>
+        </div>
+      </form>
+      ${openFormErrorDialog
+        ? html` <mwc-dialog open>
+            <div class="dialogWrapper">
+              Please correctly format email (i.e foo@bar.com).
+            </div>
+            <mwc-button
+              label="Close"
+              class="dialogButton"
+              slot="primaryAction"
+              @click="${this.toggleFormErrorDialog}"
+            ></mwc-button>
+          </mwc-dialog>`
+        : ''}
+      ${openErrorDialog
+        ? html` <mwc-dialog open>
+            <div class="dialogWrapper">
+              <div>
+                <h2>Oh no! 😭</h2>
+                <div>Unable to complete your checkout.</div>
```

Comment:
I had the thought that we should show the `status` from the returned JSON here, but I can't seem to find that value in the response payload. It's showing up as a 501, so maybe show that status? @grayside for verification on spec